l5kit/l5kit/rasterization/semantic_rasterizer.py

```diff
@@ -80,51 +62,157 @@ def __init__(
         raster_size: Tuple[int, int],
         pixel_size: np.ndarray,
         ego_center: np.ndarray,
-        semantic_map: dict,
+        semantic_map_path: str,
         pose_to_ecef: np.ndarray,
     ):
         self.raster_size = raster_size
         self.pixel_size = pixel_size
         self.ego_center = ego_center
 
         self.pose_to_ecef = pose_to_ecef
-        self.semantic_map = semantic_map
+
+        self.proto_API = MapAPI(semantic_map_path, pose_to_ecef)
+
+        self.bounds_info = self.get_bounds()
+
+    # TODO is this the right place for this function?
+    def get_bounds(self) -> dict:
+        """
+        For each elements of interest returns bounds [[min_x, min_y],[max_x, max_y]] and proto ids
+
+        Returns:
+            dict: keys are classes of elements, values are dict with `bounds` and `ids` keys
+        """
+        lanes_ids = []
+        crosswalks_ids = []
+
+        lanes_bounds = np.empty((0, 2, 2), dtype=np.float)  # [(X_MIN, Y_MIN), (X_MAX, Y_MAX)]
+        crosswalks_bounds = np.empty((0, 2, 2), dtype=np.float)  # [(X_MIN, Y_MIN), (X_MAX, Y_MAX)]
+
+        for element in self.proto_API:
+            element_id = MapAPI.id_as_str(element.id)
+
+            if self.proto_API.is_lane(element):
+                lane = self.proto_API.get_lane_coords(element_id)
+                # store bounds for fast rasterisation look-up
+                x_min = min(np.min(lane["xyz_left"][:, 0]), np.min(lane["xyz_right"][:, 0]))
+                y_min = min(np.min(lane["xyz_left"][:, 1]), np.min(lane["xyz_right"][:, 1]))
+                x_max = max(np.max(lane["xyz_left"][:, 0]), np.max(lane["xyz_right"][:, 0]))
+                y_max = max(np.max(lane["xyz_left"][:, 1]), np.max(lane["xyz_right"][:, 1]))
+
+                lanes_bounds = np.append(lanes_bounds, np.asarray([[[x_min, y_min], [x_max, y_max]]]), axis=0)
+                lanes_ids.append(element_id)
```

Comment:
Might be nice to put this into a function that does this for an element, as pretty much the same code is copy-pasted below.

Reply:
I will add a ticket for this, agree